src/marten/db/query/sql/query.cr

```diff
@@ -278,6 +297,29 @@ module Marten
             rows_affected.not_nil!
           end
 
+          private def build_average_query(column_name : String)
+            where, parameters = where_clause_and_parameters
+            limit = connection.limit_value(@limit)
+
+            sql = build_sql do |s|
+              s << "SELECT AVG(#{column_name.split(".")[-1]})"
```

Comment:
Why can't we keep the table name or table alias here? If the averaged column appears more than once in a query where multiple tables are joined (for example), this could result in errors returned by the database.

Reply:
Because we are using a subquery and the table name is lost in the outer query. We could of course use a simple query `SELECT AVG(table.column) FROM table`. Would this be sufficient?

Reply:
I don't think so; the issue is that queries making use of `LIMIT` and `OFFSET` will return no results if we don't use a subquery. Let's keep it that way for now!